app/src/main/java/org/simple/clinic/login/phone/LoginPhoneScreen.kt

```diff
@@ -73,6 +74,7 @@ class LoginPhoneScreen(context: Context, attrs: AttributeSet) : RelativeLayout(c
   }
 
   fun openLoginPinScreen() {
-    screenRouter.push(LoginPinScreen.KEY)
+    val key = screenRouter.key<LoginPhoneScreenKey>(this)
+    screenRouter.push(LoginPinScreenKey(key?.otp ?: ""))
```

Comment:
You can safely assume that `key` will never be null here.

Reply:
Then the screen router api should be changed to remote the nullable type.

Reply:
I personally try to avoid the `!!` operator unless it's happening at the java-kotlin boundaries.

Reply:
No, in this case it's impossible to hit this flow where the key will be null. Why would you still do a null check?

Reply:
TODO: Change `ScreenRouter#key()`

Reply:
Doing this on the 16-Aug PR.